modules/mono/glue/GodotSharp/GodotSharp/Core/Quaternion.cs

```diff
@@ -558,18 +558,35 @@ public Quaternion(Vector3 axis, real_t angle)
 
         public Quaternion(Vector3 arcFrom, Vector3 arcTo)
         {
-            Vector3 c = arcFrom.Cross(arcTo);
-            real_t d = arcFrom.Dot(arcTo);
-
-            if (d < -1.0f + Mathf.Epsilon)
+#if DEBUG
+            if (arcFrom.IsZeroApprox() || arcTo.IsZeroApprox())
             {
-                X = 0f;
-                Y = 1f;
-                Z = 0f;
-                W = 0f;
+                throw new ArgumentException("The vectors must not be zero.");
+            }
+#endif
+#if REAL_T_IS_DOUBLE
+            const real_t AlmostOne = 0.999999999999999;
+#else
+            const real_t AlmostOne = 0.99999975f;
+#endif
+            Vector3 n0 = arcFrom.Normalized();
+            Vector3 n1 = arcTo.Normalized();
+            real_t d = n0.Dot(n1);
+            if (Mathf.Abs(d) > AlmostOne)
+            {
+                if (d >= 0.0f)
+                {
+                    return; // Vectors are same.
```

Comment:
In the C++ code, the default values are initialized as `(0, 0, 0, 1)` for an unrotated Quaternion.

In C#, due to technical limitations, the default values are `(0, 0, 0, 0)`, so this code doesn't match the behavior of core. I believe this is the cause of this bug: https://github.com/godotengine/godot/issues/109261